sdk/storage/src/core/perform_request_response.rs

```diff
@@ -0,0 +1,23 @@
+use hyper::client::ResponseFuture;
+use url::Url;
+
+#[derive(Debug)]
+pub struct PerformRequestResponse {
+    pub(crate) url: Url,
+    pub(crate) response_future: ResponseFuture,
+}
+
+impl PerformRequestResponse {
+    pub fn url(&self) -> &Url {
```

Comment:
Can the `check_status_extract_headers_and_body` function be a method on this struct instead of a free function?

Reply:
It's a good idea, I am not sure if it's used somewhere else. I'll check.

Reply:
Turns out it's much better! 👍 👍